pkg/cmd/init.go

```diff
@@ -0,0 +1,435 @@
+				port = "<PORT>"
+				datacenters = "<DATACENTER>"
+			`),
+		},
+	}
+)
+
+type initOpts struct {
+	Provider          oneOfFlag `longflag:"provider"`
+	ClusterName       string    `longflag:"cluster-name"`
+	KubernetesVersion string    `longflag:"kubernetes-version"`
+	Terraform         bool      `longflag:"terraform"`
+	Path              string    `longflag:"path"`
+}
+
+var (
+	initProviderFlag = oneOfFlag{
+		validSet:     sets.StringKeySet(validProviders),
+		defaultValue: "none",
+	}
+)
+
+func initCmd() *cobra.Command {
+	opts := &initOpts{
+		Provider: initProviderFlag,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "init new kubeone cluster configuration",
+		Long: heredoc.Doc(`
+			Initialize new KubeOne + terraform configuration for chosen provider.
+		`),
+		SilenceErrors: true,
+		Example:       `kubeone init --provider aws`,
+		RunE: func(_ *cobra.Command, args []string) error {
+			return runInit(opts)
+		},
+	}
+
+	providerUsageText := fmt.Sprintf("provider to initialize, possible values: %s", strings.Join(opts.Provider.PossibleValues(), ", "))
+
+	cmd.Flags().BoolVar(&opts.Terraform, longFlagName(opts, "Terraform"), true, "generate terraform config")
+	cmd.Flags().StringVar(&opts.ClusterName, longFlagName(opts, "ClusterName"), "<EXAMPLE-CLUSTER>", "name of the cluster")
+	cmd.Flags().StringVar(&opts.KubernetesVersion, longFlagName(opts, "KubernetesVersion"), defaultKubeVersion, "kubernetes version")
+	cmd.Flags().StringVar(&opts.Path, longFlagName(opts, "Path"), ".", "path where to write files")
+	cmd.Flags().Var(&opts.Provider, longFlagName(opts, "Provider"), providerUsageText)
+
+	return cmd
+}
+
+func runInit(opts *initOpts) error {
+	ybuf, err := genKubeOneClusterYAML(&genKubeOneClusterYAMLParams{
+		providerName:      opts.Provider.String(),
+		clusterName:       opts.ClusterName,
+		kubernetesVersion: opts.KubernetesVersion,
+		validProviders:    validProviders,
+	})
+
+	if err != nil {
+		return fail.Runtime(err, "generating KubeOneCluster")
+	}
+
+	// special case to generate JUST yaml and no terraform
+	if opts.Path == "-" && !opts.Terraform {
+		_, err = fmt.Printf("%s", ybuf)
+
+		return err
+	}
+
+	err = os.MkdirAll(opts.Path, 0750)
+	if err != nil {
+		return err
+	}
+
+	k1config, err := os.Create(filepath.Join(opts.Path, "kubeone.yaml"))
+	if err != nil {
+		return fail.Runtime(err, "creating manifest file")
+	}
+	defer k1config.Close()
+
+	_, err = io.Copy(k1config, bytes.NewBuffer(ybuf))
+	if err != nil {
+		return fail.Runtime(err, "writing KubeOneCluster")
+	}
+
+	prov := validProviders[opts.Provider.String()]
+	if opts.Terraform && prov.terraformPath != "" {
+		if err = examples.CopyTo(opts.Path, prov.terraformPath); err != nil {
+			return fail.Runtime(err, "copying terraform configuration")
+		}
+
+		tfvars, err := os.Create(filepath.Join(opts.Path, "terraform.tfvars"))
+		if err != nil {
+			return err
+		}
+		defer tfvars.Close()
+
+		fmt.Fprintf(tfvars, "cluster_name=%q\n", opts.ClusterName)
```

Comment:
Can we just do this:

```suggestion
		fmt.Fprintf(tfvars, "cluster_name = %q\n", opts.ClusterName)
```

Some editors and tools might better parse it that way (e.g. `bat`).